src/bootstrap/download.rs

```diff
@@ -173,8 +173,7 @@ impl Config {
             // appear to have this (even when `../lib` is redundant).
             // NOTE: there are only two paths here, delimited by a `:`
             let mut entries = OsString::from("$ORIGIN/../lib:");
-            entries.push(t!(fs::canonicalize(nix_deps_dir)));
-            entries.push("/lib");
+            entries.push(t!(fs::canonicalize(nix_deps_dir)).join("lib"));
```

Comment:
This is an `OsString`, thus using `.join("lib")` here does not yield the correct result.
```suggestion
            entries.push(t!(fs::canonicalize(nix_deps_dir)).join("/lib"));
```

Reply:
[`canonicalize`](https://doc.rust-lang.org/std/fs/fn.canonicalize.html) returns a PathBuf.

Reply:
My bad, I can't read the parenthesis correctly.